### PR TITLE
[bug] Fix correct definition of torchmetrics inside pytorch lightning module

### PR DESCRIPTION
**Pull request by leoniewgnr**
## :microscope: Background
See #1361 

## :crystal_ball: Key changes
Move metrics definition inside pytorch lightning module as described in https://torchmetrics.readthedocs.io/en/latest/pages/lightning.html

### CODE REVIEW

File: neuralprophet/time_net.py [hunk 1]
```diff
@@ -163,8 +163,17 @@ def __init__(
         self.batch_size = self.config_train.batch_size
 
         # Metrics Config
+        METRICS = {
```
Comment (ourownstory) [inline neuralprophet/time_net.py:166]: We currently have `METRICS` defined in here and `utils_metrics.py`. We should only have one source for this dict. Can we consolidate this?

File: neuralprophet/utils_metrics.py [hunk 2]
```diff
@@ -22,21 +22,21 @@ def get_metrics(metric_input):
 
     Returns
     -------
-        dict
-            Dict of torchmetrics.Metric metrics.
+        dict or list
```
Comment (ourownstory) [inline neuralprophet/utils_metrics.py:25]: Maybe we should rename this function to `check_metrics`

File: neuralprophet/utils_metrics.py [hunk 3]
```diff
@@ -22,21 +22,21 @@ def get_metrics(metric_input):
...
+            return [metric_input]
         else:
             raise ValueError("Received unsupported argument for collect_metrics.")
     elif isinstance(metric_input, list):
         if all([m.upper() in METRICS.keys() for m in metric_input]):
-            return {k: v for k, v in METRICS.items() if k in metric_input}
+            return metric_input
         else:
             raise ValueError("Received unsupported argument for collect_metrics.")
     elif isinstance(metric_input, dict):
```
Comment (ourownstory) [inline neuralprophet/utils_metrics.py:42]: These user specified metrics will be defined outside and thus not be associated with a device. Might this lead to issues?

File: neuralprophet/time_net.py [hunk 4]
```diff
@@ -163,8 +163,17 @@ def __init__(
...
+            "MAE": torchmetrics.MeanAbsoluteError(),
+            "MSE": torchmetrics.MeanSquaredError(squared=True),
+            "RMSE": torchmetrics.MeanSquaredError(squared=False),
+        }
+
         self.metrics_enabled = bool(metrics)  # yields True if metrics is not an empty dictionary
         if self.metrics_enabled:
+            # only convert to dict if metrics is a list (metrics were not set in utils_metrics)
+            if isinstance(metrics, list):
+                metrics = {metric: METRICS[metric] for metric in metrics}
```
Comment (ourownstory) [inline neuralprophet/time_net.py:176]: Does moving `METRICS` to this file and transfering the objects to this dict resolve the device mismatch issue?

**Review by ourownstory — CHANGES_REQUESTED** [review]
Looking good mostly!
We just need make sure to only have one source for the list of `METRICS`. 
I also added some other open Qs

File: neuralprophet/time_net.py (hunk 4 above)
Comment (leoniewgnr) [reply neuralprophet/time_net.py:176]: yes it works now

File: neuralprophet/utils_metrics.py (hunk 3 above)
Comment (leoniewgnr) [reply neuralprophet/utils_metrics.py:43]: Resolved in newest commit

File: neuralprophet/time_net.py (hunk 1 above)
Comment (leoniewgnr) [reply neuralprophet/time_net.py]: only defined in utils_metrics.py

File: neuralprophet/utils_metrics.py (hunk 2 above)
Comment (leoniewgnr) [reply neuralprophet/utils_metrics.py]: function changed, now we are still getting metrics (only metric names, but still getting the metrics)

**Review by hxyue1 — APPROVED** [review]

**Pull request merged**